app/controllers/api/restful_controller.rb

```diff
@@ -17,6 +17,10 @@ def destroy_action
 
   private
 
+  def current_user
+    super || LoggedOutUser.new
```

Comment:
I was too afraid to change `current_user_or_visitor` to `current_user` in the rest of the app, and didn't want to write `current_user_or_visitor` everywhere in the API controller, so I remade the `current_user_or_visitor` helper here, now called `current_user` (so we can guarantee that there's always an object there for the current user)

Happy to hear different thoughts on a good way to do this.